File: app/src/main/java/com/xia/rxbus/RxBusActivity.java
package com.xia.rxbus;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.xia.flyrxbus.RxBus;
import com.xia.flyrxbus.RxBusManager;
import com.xia.flyrxbus.RxBusMessage;

import org.jetbrains.annotations.NotNull;

import androidx.appcompat.app.AppCompatActivity;

public class RxBusActivity extends AppCompatActivity {

    private TextView tvSticky;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvSticky = findViewById(R.id.tv_sticky);

        RxBus.subscribe(this, new RxBus.Callback<String>() {
            @Override
            public void onEvent(@NotNull String tag, String s) {
                tvSticky.setText(Config.appendMsg("without " + s));
                throw new NullPointerException("");
            }
        });

        RxBus.subscribe(this, "my tag", new RxBus.Callback<String>() {
            @Override
            public void onEvent(@NotNull String tag, String s) {
                tvSticky.setText(Config.appendMsg("with " + s));
            }
        });

        RxBusManager.subscribeWithTags(this, new RxBus.Callback<RxBusMessage>() {
            @Override
            public void onEvent(@NotNull String tag, RxBusMessage rxBusMessage) {
                Log.e("weixi", "tag:" + tag + "    obj:" + rxBusMessage.mObj);
                if (rxBusMessage.mObj instanceof TestEvent) {
                    final TestEvent testEvent = (TestEvent) rxBusMessage.mObj;
                    Log.e("weixi", "onEvent: " + testEvent.mString);
                }
            }
        }, "myTag1", "myTag2", "myTag3");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        RxBus.unregister(this);
    }

    public void postWithoutTag(View view) {
        Config.restoreMsg();
        RxBus.post("tag");
    }

    public void postWithTag(View view) {
        Config.restoreMsg();
        RxBus.post("with tag", "my tag");

        RxBus.post(new RxBusMessage("1"), "myTag1");
        RxBus.post(new RxBusMessage(0.001), "myTag2");
        RxBus.post(new RxBusMessage(new TestEvent("hello")), "myTag3");
    }

    public void postStickyWithoutTag(View view) {
        Config.restoreMsg();
        RxBus.postSticky("tag");
        StickyTestActivity.start(this);
    }

    public void postStickyWithTag(View view) {
        Config.restoreMsg();
        RxBus.postSticky("tag", "my tag");

        RxBus.postSticky(new RxBusMessage("1"), "myTag4");
        RxBus.postSticky(new RxBusMessage(0.001), "myTag5");
        RxBus.postSticky(new RxBusMessage(new TestEvent("hello")), "myTag6");
        StickyTestActivity.start(this);
    }

    public void useManager(View view) {
        RxBusManagerActivity.start(this);
        finish();
    }
}

File: app/src/main/java/com/xia/rxbus/RxBusActivity.kt
package com.xia.rxbus

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.xia.flyrxbus.RxBus
import com.xia.flyrxbus.RxBusManager
import com.xia.flyrxbus.RxBusMessage
import kotlinx.android.synthetic.main.activity_main.*

class RxBusActivity : AppCompatActivity() {

    private var mTvSticky: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mTvSticky = findViewById(R.id.tv_sticky)

        RxBusManager.subscribe(this, object : RxBus.Callback<String>() {
            override fun onEvent(tag: String, s: String) {
                mTvSticky!!.text = Config.appendMsg("without $s")
                throw NullPointerException("")
            }
        })

        RxBusManager.subscribe(this, "my tag", object : RxBus.Callback<String>() {
            override fun onEvent(tag: String, s: String) {
                mTvSticky!!.text = Config.appendMsg("with $s")
            }
        })

        RxBusManager.subscribeWithTags(this, object : RxBus.Callback<RxBusMessage>() {
            override fun onEvent(tag: String, rxBusMessage: RxBusMessage) {
                if ("myTag1" == tag) {
                    Log.e("weixi", "tag:" + tag + "    obj:" + rxBusMessage.mObj)
                }
                if ("myTag2" == tag) {
                    Log.e("weixi", "tag:" + tag + "    obj:" + rxBusMessage.mObj)
                }
                if ("myTag3" == tag) {
                    Log.e("weixi", "tag:" + tag + "    obj:" + rxBusMessage.mObj)
                    if (rxBusMessage.mObj is TestEvent) {
                        val testEvent = rxBusMessage.mObj as TestEvent
                        Log.e("weixi", "tag:" + tag + "    onEvent: " + testEvent.mString)
                    }
                }
            }
        }, "myTag1", "myTag2", "myTag3")

        btn_post_without_tag.setOnClickListener {
            Config.restoreMsg()
            RxBusManager.post("tag")
        }

        btn_post_with_tag.setOnClickListener {
            Config.restoreMsg()
            RxBusManager.post("with tag", "my tag")
            RxBusManager.post(RxBusMessage("1"), "myTag1")
            RxBusManager.post(RxBusMessage(0.001), "myTag2")
            RxBusManager.post(RxBusMessage(TestEvent("hello")), "myTag3")
        }

        btn_post_sticky_without_tag.setOnClickListener {
            Config.restoreMsg()
            RxBusManager.postSticky("tag")
            StickyTestActivity.start(this)
        }

        btn_post_sticky_with_tag.setOnClickListener {
            Config.restoreMsg()
            RxBusManager.postSticky("tag", "my tag")
            RxBusManager.postSticky(RxBusMessage("1"), "myTag4")
            RxBusManager.postSticky(RxBusMessage(0.001), "myTag5")
            RxBusManager.postSticky(RxBusMessage(TestEvent("hello")), "myTag6")
            StickyTestActivity.start(this)
        }

        btn_use_manager.setOnClickListener {
            RxBusManagerActivity.start(this)
            finish()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        RxBusManager.unregister(this)
    }
}
